crest/Assets/Crest/Crest/Scripts/LodData/RegisterLodDataInput.cs

```diff
@@ -26,15 +43,16 @@ public abstract class RegisterLodDataInputBase : MonoBehaviour, ILodDataInput
 
         public static int sp_Weight = Shader.PropertyToID("_Weight");
 
-        static Dictionary<System.Type, List<ILodDataInput>> _registrar = new Dictionary<System.Type, List<ILodDataInput>>();
+        static DuplicateKeyComparer<int> s_comparer = new DuplicateKeyComparer<int>();
+        static Dictionary<Type, OceanInput> s_registrar = new Dictionary<Type, OceanInput>();
 
-        public static List<ILodDataInput> GetRegistrar(System.Type lodDataMgrType)
+        public static OceanInput GetRegistrar(Type lodDataMgrType)
```

Comment:
Iterating over a sorted list in a foreach loop generates a lot of garbage.

Reply:
Thanks good catch